packages/carbon-graphs/src/js/controls/Bar/Bar.js

```diff
@@ -34,22 +34,54 @@ import {
   translateTextLabel,
 } from './helpers/translateHelpers';
 
+/**
+ * @typedef {object} Bar
+ * @typedef {object} GraphContent
+ * @typedef {object} BarConfig
+ */
 /**
  * Calculates the min and max values for Y Axis or Y2 Axis
  *
  * @private
  * @param {Array} values - Datapoint values
- * @param {string} axis - y or y2
  * @returns {object} - Contains min and max values for the data points
  */
-const calculateValuesRange = (values, axis = constants.Y_AXIS) => {
+const calculateValuesRangeYAxis = (values) => {
   const min = Math.min(...values.map((i) => i.y));
   const max = Math.max(...values.map((i) => i.y));
   return {
-    [axis]: {
-      min: min < 0 ? min : 0,
-      max: max > 0 ? max : 0,
-    },
+    min: min < 0 ? min : 0,
+    max: max > 0 ? max : 0,
+  };
+};
+
+/**
+ * @typedef {object} Bar
+ * @typedef {object} GraphContent
+ * @typedef {object} BarConfig
+ */
+/**
+ * Calculates the min and max values for X-axis.
+ * First we filter out values that are `null`, this is a result of
+ * datapoint being part of being in a non-contiguous series and then we
+ * get the min and max values for the X-axis domain.
+ *
+ * @private
+ * @param {Array} values - Datapoint values
+ * @returns {object} - Contains min and max values for the data points for x-axis
+ */
+const calculateValuesRangeXAxis = (values) => {
+  const xAxisValuesList = values.filter((i) => i.x !== null && i.x !== undefined).map((i) => {
+    // if the x-axis is a timeseries, then convert it to an epoc int
+    // for easier calculations
+    if (typeof i.x === 'string' || i.x instanceof Date) {
+      return utils.getEpocFromDateString(i.x);
+    }
+    return i.x;
+  });
+  return {
+    min: Math.min(...xAxisValuesList),
+    max: Math.max(...xAxisValuesList),
   };
```

Comment:
I would also update this to match https://github.com/cerner/terra-graphs/blob/0db73f925acd20425bf93ab5f49c1b162126126d/packages/carbon-graphs/src/js/controls/Bubble/helpers/helpers.js#L510-L530

Reply:
Fixed [ab6c070](https://github.com/cerner/terra-graphs/pull/293/commits/ab6c0704180e22a19172a6ba2d7663360a47002a)